packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/Credentials.kt

```diff
@@ -131,5 +132,18 @@ public interface Credentials {
         public fun jwt(jwtToken: String): Credentials {
             return CredentialsImpl(CredentialsImpl.jwt(jwtToken))
         }
+
+        /**
+         * Creates credentials representing a login using an App Services Function. The payload would
+         * be serialized and parsed as an argument to the remote function. It is required to be
+         * a [Map] or [BsonDocument]. The payload keys must match the format and names the function expects.
+         *
+         * @param payload The payload that will be passed as an argument to the server function.
+         * @return a set of credentials that can be used to log into an App Services Application
+         * using [App.login].
+         */
+        public fun customFunction(payload: Any): Credentials {
```

Comment:
Given the current restrictions, should we just create two functions `customFunction(payload: Map<String, Any?>)` and `customFunction(payload: BsonDocument)`....then when/if we add support for full serialization, we could combine them into `customFunction(payload: Any)`?